internal/pkg/certs/ca.go

```diff
@@ -10,21 +10,18 @@ import (
 	"time"
 )
 
-type CA struct {
-	CaCert       []byte
-	CaPrivateKey *rsa.PrivateKey
-}
-
-func GenerateCA() (*CA, error) {
-	privateKey, err := newPrivateKey(1024)
+func GenerateCA() ([]byte, *rsa.PrivateKey, error) {
```

Comment:
Can you add some inline documentation explaining what is returned by the function?